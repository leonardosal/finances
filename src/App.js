import React from 'react';

import Home from './containers/Home';
import Header from './components/Header';

import './styles/global.scss';

const App = () => (
  <>
    <Header />
    <Home />
  </>
);

export default App;
