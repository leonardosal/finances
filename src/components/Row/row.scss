.row {
  display: flex;
  justify-content: space-between;
  padding: 16px;
  border-bottom: 1px solid #eee;
  font-size: 1rem;
  @media (min-width: 1025px) {
    font-size: 1rem;
  }

  .credit {
    color: #04a11e;
  }
  .debit {
    color: #e62705;
  }
}
