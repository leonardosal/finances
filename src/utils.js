import uuidv4 from 'uuid/v4';

export const TYPE = {
  DEBIT: 'DEBIT',
  CREDIT: 'CREDIT',
};

export const formatValue = value => {
  return parseInt(value, 10).toLocaleString('pt-br', {
    minimumFractionDigits: 2,
    style: 'currency',
    currency: 'BRL',
  });
};

export const calculateBalance = transactions => {
  return transactions.reduce((acc, current) => {
    if (current.type === TYPE.DEBIT) {
      return parseInt(acc, 10) - parseInt(current.value, 10);
    }
    return parseInt(acc, 10) + parseInt(current.value, 10);
  }, 0);
};

export const saveInStorage = transactions => {
  window.localStorage.setItem('@transactions', JSON.stringify(transactions));
};

export const loadFromStorage = () => {
  const strTransactions = window.localStorage.getItem('@transactions');
  const transactions = JSON.parse(strTransactions);
  return transactions;
};

export const addItem = (list, item) => {
  return [{ id: uuidv4(), ...item }, ...list];
};
