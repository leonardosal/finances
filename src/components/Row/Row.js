import React from 'react';
import propTypes from 'prop-types';

import './row.scss';

import { formatValue } from '../../utils';

const Row = ({ description, value, type }) => (
  <div className="row">
    <span>{description}</span>
    <span className={type === 'DEBIT' ? 'debit' : 'credit'}>
      {formatValue(value)}
    </span>
  </div>
);

Row.propTypes = {
  description: propTypes.string,
  type: propTypes.string,
  value: propTypes.number,
};

Row.defaultProps = {
  description: '',
  type: 'CREDIT',
  value: 0,
};

export default Row;
