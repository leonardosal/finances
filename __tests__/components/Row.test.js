import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Row from '../../src/components/Row';
import { TYPE } from '../../src/utils';

it('should render Row component with correct data props for expense', () => {
  const props = {
    description: 'Uber',
    type: TYPE.DEBIT,
    value: 17.0,
  };
  const component = mount(
    <Row
      description={props.description}
      type={props.type}
      value={props.value}
    />
  );
  expect(toJson(component)).toMatchSnapshot();
});

it('should render Row component with correct data props for revenue', () => {
  const props = {
    description: 'Salary',
    type: TYPE.CREDIT,
    value: 2000.0,
  };
  const component = mount(
    <Row
      description={props.description}
      type={props.type}
      value={props.value}
    />
  );
  expect(toJson(component)).toMatchSnapshot();
});
