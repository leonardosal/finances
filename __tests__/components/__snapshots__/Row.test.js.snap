// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Row component with correct data props for expense 1`] = `
<Row
  description="Uber"
  type="DEBIT"
  value={17}
>
  <div
    className="row"
  >
    <span>
      Uber
    </span>
    <span
      className="debit"
    >
      R$17,00
    </span>
  </div>
</Row>
`;

exports[`should render Row component with correct data props for revenue 1`] = `
<Row
  description="Salary"
  type="CREDIT"
  value={2000}
>
  <div
    className="row"
  >
    <span>
      Salary
    </span>
    <span
      className="credit"
    >
      R$2.000,00
    </span>
  </div>
</Row>
`;
